### PR TITLE
Update TCK user guide to build on JDK11+

### PR DESCRIPTION
**Pull request by tvallin**
Signed-off-by: tvallin <thibault.vallin@oracle.com>

### CODE REVIEW

File: user_guides/saaj/src/main/jbake/content/toc.adoc [hunk 1]
```diff
@@ -63,9 +63,9 @@ link:install.html#GBFTP[3 Installation]
 link:config.html#GBFVV[4 Setup and Configuration]
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-** link:config.html#GHGDH[4.1 Configuring Your Environment to Run the TCK With a Compatible Implementation]
-** link:config.html#GHGDH[4.2 Configuring Your Environment to Run the TCK With Your Concurreny Implementation]
-** link:config.html#GHGDH[4.3 Publishing the Test Applications]
+** link:config.html#GBFVU[4.1 Configuring Your Environment to Run the TCK With a Compatible Implementation]
+** link:config.html#GCLHU[4.2 Configuring Your Environment to Run the TCK With Your Concurreny Implementation]
+** link:config.html#GHGDG[4.3 Publishing the Test Applications]
```
Comment (lukasj) [inline user_guides/saaj/src/main/jbake/content/toc.adoc:68]: in general - `toc.adoc` files are being created by the build, so ideally - existing ones should be deleted/removed from the repo, `clean` goal updated to delete this file during `mv clean` and then added to `.gitignore` to not accidentally commit these files again

Comment (tvallin) [reply user_guides/saaj/src/main/jbake/content/toc.adoc]: Thanks for extra explanation.

**Comment by RohitKumarJain** [comment]
Hi @tvallin , as per my understanding this is 9.1 release. If you look at the Jakarta-EE-TCK-Users-Guide.pdf file present under https://ci.eclipse.org/jakartaee-tck/job/buildUG/lastSuccessfulBuild/artifact/user_guides/jakartaee/target/generated-docs/, location, there is a "common note" in each page which is "Jakarta EE Platform, Enterprise Edition 9 Test Compatibility Kit User’s Guide, Release 9 for Jakarta EE". 

- I believe this should be "Jakarta EE Platform, Enterprise Edition 9 Test Compatibility Kit User’s Guide, Release 9.1 for Jakarta EE". As the current release is the 9.1 release.
- Also, I noticed on each alternative page, there is some alignment issue. Ex: This common note is starting from the Left most corner on even-numbered pages (ex: Page# 2,4,6,8... so on) while on odd-numbered pages (ex: Page# 1,3,5,7... so on) the same common note is starting from the middle and heading towards the right most corner.

Note: I am not really sure if this is expected behaviour or need enhancement.

**Comment by lukasj** [comment]
@RohitKumarJain the intent of this PR is to allow the build of UGs on JDK 11, which currently does not work at all. If there are issues in UGs themselves, like version numbers or consistency related issues, different issues/PRs should be created to cover them. It, IMHO, does not make sense to try to fix everything at once as that gives small room for collaborative and iterative work.

**Comment by RohitKumarJain** [comment]
> @RohitKumarJain the intent of this PR is to allow the build of UGs on JDK 11, which currently does not work at all. If there are issues in UGs themselves, like version numbers or consistency related issues, different issues/PRs should be created to cover them. It, IMHO, does not make sense to try to fix everything at once as that gives small room for collaborative and iterative work.

I agree with you @lukasj, I was just reviewing and shared my inputs. We can definitely consider these changes in another PR.

**Comment by tvallin** [comment]
Hello @RohitKumarJain,
I agree for the release number, but about the alignment issue, I think it is meant to be this way because of recto verso pages.
Anyway, if you notice any issue(s), feel free to open an issue in the issue tracker.

**Comment by lukasj** [comment]
I've filed #661 to update version number(s)

**Comment by RohitKumarJain** [comment]
> I've filed #661 to update version number(s)

Thank you @lukasj

**Comment by scottmarlow** [comment]
Has anyone built this branch and verified that any changed TCK User Guides (html + pdf) have the expected content?   

At the end of the EE 9 release, we noticed that after making several TCK User Guide changes, that the generated pdf was missing a lot of content (e.g. chapters were empty).  It seems that the build tooling doesn't catch the syntax failures, so we have to visually examine the html/pdf output from the TCK build jobs.

**Comment by lukasj** [comment]
@scottmarlow is it known what UGs were/are affected? I noticed that what you say could happen _if_ the toc file was committed in the repo instead of being generated during the build. Other cause could be wrong content/ordering of inc files. Allowing building UGs on JDK 11 is the only intent of this PR. Nothing else - no touch of the UGs' content is planned. Or do you expect adding support for build on JDK 11 fixes all issues in all touched UGs as well?

This change is supposed to be the same as https://github.com/eclipse-ee4j/jaf-tck/pull/24, https://github.com/eclipse-ee4j/mail-tck/pull/19 or https://github.com/eclipse-ee4j/jaxb-tck/pull/50

**Comment by gurunrao** [comment]
> Has anyone built this branch and verified that any changed TCK User Guides (html + pdf) have the expected content?
> 
> At the end of the EE 9 release, we noticed that after making several TCK User Guide changes, that the generated pdf was missing a lot of content (e.g. chapters were empty). It seems that the build tooling doesn't catch the syntax failures, so we have to visually examine the html/pdf output from the TCK build jobs.

@scottmarlow - Had built UG with PR branch here - https://ci.eclipse.org/jakartaee-tck/job/buildUG/18/
@RohitKumarJain is reviewing for generated documents for any alignment issue. @RohitKumarJain  - Please add your inputs to the PR.

**Comment by scottmarlow** [comment]
> @scottmarlow is it known what UGs were/are affected? 
I think it was just the Jakarta EE Platform TCK user guide which @alwin-joseph corrected.  

> I noticed that what you say could happen _if_ the toc file was committed in the repo instead of being generated during the build. Other cause could be wrong content/ordering of inc files. 
Good to know, thanks!

> Allowing building UGs on JDK 11 is the only intent of this PR. Nothing else - no touch of the UGs' content is planned. 
I'm not worried about touching more than the intent of this PR, just that we verify that the resulting .pdf + .html files contain the expected content (e.g. not that we need to verify word for word, instead just a visual verification that each section has a reasonable amount of text to prove the doc build worked).

> Or do you expect adding support for build on JDK 11 should also cover fixing all/the most visible issues in all touched UGs as well?
I'm not against making such changes to improve UGs, as long as someone has time to review the changes.

> 
> This change is supposed to be the same as [eclipse-ee4j/jaf-tck#24](https://github.com/eclipse-ee4j/jaf-tck/pull/24), [eclipse-ee4j/mail-tck#19](https://github.com/eclipse-ee4j/mail-tck/pull/19) or [eclipse-ee4j/jaxb-tck#50](https://github.com/eclipse-ee4j/jaxb-tck/pull/50)

:+1:

**Comment by scottmarlow** [comment]
> > Has anyone built this branch and verified that any changed TCK User Guides (html + pdf) have the expected content?
> > At the end of the EE 9 release, we noticed that after making several TCK User Guide changes, that the generated pdf was missing a lot of content (e.g. chapters were empty). It seems that the build tooling doesn't catch the syntax failures, so we have to visually examine the html/pdf output from the TCK build jobs.
> 
> @scottmarlow - Had built UG with PR branch here - https://ci.eclipse.org/jakartaee-tck/job/buildUG/18/
> @RohitKumarJain is reviewing for generated documents for any alignment issue. @RohitKumarJain - Please add your inputs to the PR.

:+1:

**Review by alwin-joseph — APPROVED** [review]
As the PR is intended to build the UG on JDK11, I was able to verify the same works.

**Comment by edbratt** [comment]
After updating with this change, the PDF render isn't finding the embedded images in the Jakarta EE TCK User's Guide. You can reproduce this by checking out the current head and just building that document. I am using Java 8 (java version "1.8.0_191")
(cd user_guides/jakartaee; mvn clean install)
View the PDF produced in target/generated-docs/Jakarta-EE-TCK-Users-Guide.pdf
Compare that against the documents produced in the HTML render in target/staging.
Specifically, look at Intro section, image 1.1 (target/staging/intro.html) Figure 1-1 Typical Jakarta Platform, Enterprise Edition Workflow. In the PDF, you will just find the alternate text.
This comes from src/main/jbake/content/intro.adoc

Does this work correctly for you ?

Any ideas for how I might address this? This is important as we want to generate RC copies of the TCKs and the user guides are part of that.

**Comment by lukasj** [comment]
@edbratt #705 should fix the problem in the platform tck pdf guide

@tvallin can you check other guides, apply the fix where appropriate and submit PR, please? Thx!

**Comment by tvallin** [comment]
Sure, I am on it !

**Comment by edbratt** [comment]
I updated the pom.xml in my clone and the PDF works just like before. THANK YOU! (I will include that change in my commit. Sorry, I'm not sure how to weave these together more accurately.)

**Comment by gurunrao** [comment]
@tvallin - we see build issue with JDK 8, Link for build logs can be found here -  https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck/job/master/1393/artifact/build_jakartaeetck.log

**Comment by tvallin** [comment]
@gurunrao #708 should fix the build issue.

File: user_guides/jacc/pom.xml [hunk 2]
```diff
@@ -43,38 +43,40 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <site.output.dir>${project.build.directory}/staging</site.output.dir>
         <maven.site.skip>true</maven.site.skip>
+        <maven.deploy.skip>true</maven.deploy.skip>
```
Comment (pzygielo) [inline user_guides/jacc/pom.xml:46]: Duplicate of line 56

Comment (tvallin) [reply user_guides/jacc/pom.xml:46]: PR created #709

**Comment by gurunrao** [comment]
Platform TCK build is success now - https://ci.eclipse.org/jakartaee-tck/job/9.1/job/eftl-jakartaeetck-build-910/lastBuild/

**Comment by scottmarlow** [comment]
Thanks @tvallin for the contribution!  Thanks to others here that helped! :-)

**Pull request merged**